src/continuous_agg.c

```diff
@@ -404,6 +404,33 @@ ts_continuous_aggs_get_minimum_invalidation_time(int64 modification_time,
 	}
 }
 
+/* Find a continuous aggregate by the materialized hypertable id */
+ContinuousAgg *
+ts_continuous_agg_find_by_mat_hypertable_id(int32 mat_hypertable_id)
+{
+	ContinuousAgg *ca = NULL;
+	ScanIterator iterator =
+		ts_scan_iterator_create(CONTINUOUS_AGG, RowExclusiveLock, CurrentMemoryContext);
+
+	/* Note that this scan can only match at most once */
+	init_scan_by_mat_hypertable_id(&iterator, mat_hypertable_id);
+	ts_scanner_foreach(&iterator)
+	{
+		bool should_free;
+		HeapTuple tuple = ts_scan_iterator_fetch_heap_tuple(&iterator, false, &should_free);
+		Form_continuous_agg form = (Form_continuous_agg) GETSTRUCT(tuple);
+		ca = ts_scan_iterator_alloc_result(&iterator, sizeof(*ca));
```

Comment:
It's nice to separate declaration bock from the body for readability:
```suggestion

		ca = ts_scan_iterator_alloc_result(&iterator, sizeof(*ca));
```